### PR TITLE
Sankey plot

### PR DESCRIPTION
**Pull request by mabudz**
# Please confirm that this PR has done the following:

- ~Tests Added~
- [x] Documentation Added
- [x] Name of contributors Added to AUTHORS.rst
- [x] Description in RELEASE_NOTES.md Added

# Description of PR

This PR introduces a function sankey() to create a basic Sankey diagram using plotly [https://plotly.com/python/sankey-diagram/].  This figure can be further modified (e.g. `fig.update_traces()`) or used (e.g. `fig.write_html()`) by the user [https://plotly.com/python/graph-objects/]

The function uses a  mapping dictionary {'variable1': ('source1', 'target1'), ...} to specify the source and the target node for the considered variables.

Currently only one region and one year can be plotted.

### CODE REVIEW

**Comment by danielhuppmann** [comment]
there was an update of the sphinxcontrib-bibtex package and a change of the WorldBank data used in the pandas_datarader test, which currently causes unit tests to fail - don't worry about that for now

**Comment by mabudz** [comment]
Alright. I just address the recommendations of picky Mr.Sticklers

**Review by danielhuppmann — CHANGES_REQUESTED** [review]
thanks @mabudz for this nice addition! a few suggestions to clean up the implementation... Also, I resolved all issues by stickler that you have implemented, but there are still a few stickler requests...

**Comment by danielhuppmann** [comment]
I also just realized that you need to add `plotly` to the dependencies explicitly, i.e. [here](https://github.com/IAMconsortium/pyam/blob/a7a5f4f346ac4327678993db64aa3b91b596daf7/setup.py#L14)

**Comment by mabudz** [comment]
Ah, I forgot to add the plotly dependencies. Should not do this while sitting on the airport :)))

File: pyam/figures.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        df : pd.DataFrame
+            data to plot as wide format
+        mapping : dict
+            Assigns the source and target component of a variable
+
+            .. code-block:: python
+
+            {
+                variable: (source, target),
+                }
```
Comment (danielhuppmann) [inline pyam/figures.py:25]: this section has to be indented (beyond the `.. code-block::`) to be displayed correctly
```suggestion
                {
                    variable: (source, target),
                }
```

**Review by danielhuppmann — APPROVED** [review]
One minor suggestion for a formatting clean-up inline, and please add to the release notes - then good to go! As discussed, we'll tackle an example for the plotting gallery and tests during/after the holidays...

**Comment by danielhuppmann** [comment]
Merging and tackling the tests in a next iteration - thanks @mabudz!

**Pull request merged**